### PR TITLE
feat: cp-8368 historyservice

### PR DESCRIPTION
**Pull request by vvava**
## Description

We want to use to fetch the transaction histories via the vm-modules.

## Changes

Get rid of the old solutions and a unified history service fetch all the activities through the modules system.

## Testing

Onboarding with a wallet which has p and / or x chain history. Try to go to all the chains and check the Activity, it should be almost the same as before except the ERC-1155 token transfers should show up.

## Screenshots:

https://github.com/user-attachments/assets/650b2c3d-0879-4e90-8760-9f88bfbff0f4

## Checklist for the author

Tick each of them when done or if not applicable.

- [x] I've covered new/modified business logic with Jest test cases.
- [x] I've tested the changes myself before sending it to code review and QA.

### CODE REVIEW

**Review by meeh0w — COMMENTED** [review]
I know it's still a draft, but just wanted to leave my thoughts:

I liked how @gergelylovas got rid of different types of balances services and only left one `BalancesService`. Can we do the same with the `HistoryService`s?

For reference, here's the link to his `BalancesService`:
* https://github.com/ava-labs/core-extension/pull/43/files#diff-fd411839e9eef1b07f0b63cedd78196b77fdca273ec67f9d25a54070ba6435d8

We could essentially be left with just `HistoryService` here that could load specific module for a given network and invoke `getTransactionHistory()` on it.

File: src/pages/Wallet/components/History/components/ActivityCard/ActivityCardIcon.tsx [hunk 1]
```diff
@@ -50,7 +48,20 @@ export function ActivityCardIcon({ historyItem }: ActivityCardProp) {
       <ArrowDownIcon size={iconSize} />
     );
 
-    switch (historyItem.type) {
+    if (historyItem.isBridge) {
+      setTxIcon(<BridgeIcon size={iconSize} />);
+      return;
+    }
+
+    // TODO: check the vmmodules for the type
```
Comment (meeh0w) [inline src/pages/Wallet/components/History/components/ActivityCard/ActivityCardIcon.tsx:56]: Is this comment still needed?

File: src/pages/Wallet/WalletRecentTxs.tsx [hunk 2]
```diff
@@ -219,19 +213,21 @@ export function WalletRecentTxs({
...
       return true;
     } else if (filter === FilterType.BRIDGE) {
-      return tx.isBridge;
+      return tx.txType === TransactionType.BRIDGE;
     } else if (filter === FilterType.SWAP) {
-      return tx.type === TransactionType.SWAP;
+      return tx.txType === TransactionType.SWAP;
     } else if (filter === FilterType.CONTRACT_CALL) {
-      return tx.isContractCall && tx.type !== TransactionType.SWAP;
+      return tx.isContractCall && tx.txType !== TransactionType.SWAP;
```
Comment (meeh0w) [inline src/pages/Wallet/WalletRecentTxs.tsx:220]:
```suggestion
      return tx.isContractCall && !tx.isBridge && tx.txType !== TransactionType.SWAP;
```

File: src/pages/Wallet/WalletRecentTxs.tsx [hunk 3]
```diff
@@ -219,19 +213,21 @@ export function WalletRecentTxs({
     if (filter === FilterType.ALL) {
       return true;
     } else if (filter === FilterType.BRIDGE) {
-      return tx.isBridge;
+      return tx.txType === TransactionType.BRIDGE;
```
Comment (meeh0w) [inline src/pages/Wallet/WalletRecentTxs.tsx:216]:
```suggestion
      return tx.isBridge || tx.txType === TransactionType.BRIDGE;
```

**Review by meeh0w — CHANGES_REQUESTED** [review]
I also took for tor a quick test and found filtering for `Bridge` in activity tab doesn't seem to work. Posted suggestions for fixes below :)

**Review by meeh0w — APPROVED** [review]

**Review by gergelylovas — APPROVED** [review]

**Pull request merged**